sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -371,8 +450,16 @@ static void rt1308_init(const struct snd_soc_acpi_link_adr *link,
 		return;
 
 	rt1308_num++;
-	if (rt1308_num > 1)
-		dai_links->init = second_spk_init;
+	if (rt1308_num == 1)
+		dai_links->init = first_spk_init;
+
+	if (rt1308_num == 2) {
+		/* two 1308 in one dai link */
+		if (dai_links->init)
+			dai_links->init = all_spk_init;
+		else
+			dai_links->init = second_spk_init;
```

Comment:
wow, this is very odd code, please explain what you are doing. How exactly do you detect that there are two devices on the same link?